Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+// ===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// ===----------------------------------------------------------------------===//
+
+import Dispatch
+import Foundation
+import PackagePlugin
+
+@main
+struct AWSLambdaPackager: CommandPlugin {
+    func performCommand(context: PackagePlugin.PluginContext, arguments: [String]) async throws {
+
+        let configuration = try Configuration(context: context, arguments: arguments)
+        guard !configuration.products.isEmpty else {
+            throw Errors.unknownProduct("no appropriate products found to deploy")
+        }
+
+        let samExecutable = try context.tool(named: "sam")
+        let samExecutableUrl = URL(fileURLWithPath: samExecutable.path.string)
```

Comment:
not sure we need a URL here. we can probably use Path from the PackagePlugin API

Reply:
`URL` is required here because I use `String.write(to:encoding:)` to save the generated deployment descriptor

Reply:
`String.write(to:encoding:)` is syntactic sugar for https://developer.apple.com/documentation/foundation/filemanager/1410695-createfile which takes strings

Reply:
done https://github.com/sebsto/swift-aws-lambda-runtime/commit/d09ecbe35bc8cb2c72f3afffd5dec4d3100f87dd